Components/buttons.css

```diff
@@ -0,0 +1,110 @@
+:root{
+    --btn-violet-dark : rgb(129, 7, 129);
+    --btn-violet-light: rgb(145, 59, 145);
+    --btn-solid-hover-bg : rgb(180, 10, 180);
+    --btn-white : white;
+    --btn-black : black;
+    --btn-box-shadow : 1px 4px 10px 1px rgba(59, 63, 66, 0.4);
+}
+
+.btn-solid{
+    display: inline-block;
+    text-decoration: none;
+    text-align: center;
+    outline: none;
+    background: var(--btn-violet-dark);
+    color: var(--btn-white);
+    cursor: pointer;
+    border: none;
+    box-shadow: var(--btn-box-shadow);
+    border-radius: 6px;
+}
+
+.btn-solid:hover{
+    background: var(--btn-solid-hover-bg);
+    transform: scale(1.02);
+}
+
+.btn-outline{
+    display: inline-block;
+    text-decoration: none;
+    text-align: center;
+    outline: none;
+    background: var(--btn-white);
+    color: var(--btn-violet-dark);
+    cursor: pointer;
+    border: 2px solid var(--btn-violet-dark) ;
+    box-shadow: var(--btn-box-shadow);
+    border-radius: 6px;
+}
+
+.btn-outline:hover{
+    background: var(--btn-violet-dark);
+    color: var(--btn-white);
+    transform: scale(1.01);
+}
+
+.btn-link{
+   color: var(--btn-black);
+   font-size: large;
+}
+
+.btn-link:hover{
+    color: var(--btn-violet-dark);  
+}
+
+.btn-icon{
+    background: transparent;
+    border: transparent;
+    font-size: large;
+    color: var(--btn-violet-dark);
+    cursor: pointer;
+}
+
+.btn-icon:hover{
+    color: var(--btn-violet-light);
+}
+
+.btn-icon-text{
+    background: transparent;
+    border: transparent;
+    font-size: medium;
+    padding: 0.5rem 1rem;
+    background-color: var(--btn-violet-dark);
+    color: var(--btn-white);
+    cursor: pointer;
+    border-radius: 5px;
+}
+
+.btn-icon-text:hover{
+    background-color: var(--btn-violet-light);
+}
+
+.button-floating{
+    background: transparent;
+    border: transparent;
+    color: var(--btn-violet-light);
+    cursor: pointer;
```

Comment:
This won't work as a floating button. The floating button has a `position: fixed` and a higher z-index.
(You can check my code for reference)

Note: you need to add `position: static` using a different class so that it stays in your `action-container` for the documentation page.

Reply:
1. z-index part I agree .
2. If I give position : fixed the button would come out of the container and would be placed with respect to viewport so are you suggesting me to use position : static instead of fixed so that it stays inside the container. Is that what you meant?